core/templates/dev/head/pages/exploration_editor/simple_editor_tab/SimpleEditorManagerService.js

```diff
@@ -138,6 +140,37 @@ oppia.factory('SimpleEditorManagerService', [
         data.questionList.addQuestion(QuestionObjectFactory.create(
           lastStateName, stateData.interaction, ''));
       },
+      changeQuestion: function(type, index) {
+        var currentStateName = (
+          index === 0 ? 'Introduction' : 'Question ' + index);
+        var nextStateName = 'Question ' + (index + 1);
+        var currentInteractionId = (
+          data.questionList._questions[index]._interactionId);
+
+        // Update Question Type If interactionId is not same.
+        if(type != currentInteractionId) {
+          var newAnswerGroups = [];
+          if(type === 'MultipleChoiceInput') {
```

Comment:
Since this might extend to many interactions, it might be better if we made this as a `case` clause.